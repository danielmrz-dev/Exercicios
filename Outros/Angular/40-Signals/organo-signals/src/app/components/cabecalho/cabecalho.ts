import { Component } from '@angular/core';

@Component({
  selector: 'app-cabecalho',
  imports: [],
  templateUrl: './cabecalho.html',
  styleUrl: './cabecalho.scss'
})
export class Cabecalho {

}
