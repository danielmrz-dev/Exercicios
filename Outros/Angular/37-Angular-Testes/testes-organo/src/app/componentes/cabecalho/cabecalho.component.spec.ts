import { CabecalhoComponent } from "./cabecalho.component";
import { ComponentFixture, TestBed } from '@angular/core/testing';

describe('Description', () => {
    let component: CabecalhoComponent;
    let fixture: ComponentFixture<CabecalhoComponent>

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [CabecalhoComponent],
        })
        fixture = TestBed.createComponent(CabecalhoComponent);
        component = fixture.componentInstance;
    });
    

    it('deve ser criado', () => {
        expect(component).toBeTruthy();
    });

    it('deveria definir as propriedades alt e src', () => {
        expect(component.alt).toBeDefined();
        expect(component.src).toBeDefined();
    });

    it('deveria renderizar o conteÃºdo baseado nas propriedades src e alt', () => {
        component.src = "https://example.com/test-image.jpg";
        component.alt = "Teste";
        expect(component).toMatchSnapshot();
    });
    
    
});