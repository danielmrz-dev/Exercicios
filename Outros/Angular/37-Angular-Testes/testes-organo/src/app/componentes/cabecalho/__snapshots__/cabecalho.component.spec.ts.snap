// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description deveria renderizar o conte√∫do baseado nas propriedades src e alt 1`] = `
CabecalhoComponent {
  "__ngContext__": 4,
  "alt": "Teste",
  "src": "https://example.com/test-image.jpg",
}
`;
